### ISSUE TITLE
Failed to Create DirectSound Device

### ISSUE THREAD

**Issue by time-killer-games** [enigma-dev/enigma-dev#1770]
The title of this issue quotes an error message DirectSound shows on startup of a game before crashing directly after if the user doesn't have any speakers plugged in.

**Comment by time-killer-games** [comment]
also happens when there are speakers plugged in but they aren't being recognized for whatever reason, if I had to guess.

**Comment by RobertBColton** [comment]
I think I knew about this problem or read about it before, it just never happened until now. What does GM8 do in this case? I am not interested in what GMS does since it uses OpenAL if new audio system is checked.

https://github.com/enigma-dev/enigma-dev/blob/82ad7398449ff7c26d66bf9d4c9b6b7448fcbb7d/ENIGMAsystem/SHELL/Audio_Systems/DirectSound/DSsystem.cpp#L84

**Comment by time-killer-games** [comment]
GM 8.1 runs the game regardless, same with GMStudio 1.4 and 2.x.

**Comment by time-killer-games** [comment]
This same thing happens with OpenAL. It errors a similar error. But there is one difference, it doesn't crash after the error in OpenAL, at least...

**Comment by RobertBColton** [comment]
Actually, I discovered GM8.1 displays the following error message which can not be dismissed.
![GM8.1 Audio Device Error Message](https://user-images.githubusercontent.com/3212801/85426484-844af880-b548-11ea-8bc5-c8ea2c695509.png)

**Comment by time-killer-games** [comment]
lol yoyogames seriously put wtf in the title bar

**Comment by RobertBColton** [comment]
Yeah so this is definitely documented and surprisingly I can now reproduce it using OpenAL too.
https://docs.microsoft.com/en-us/previous-versions/windows/desktop/ee416365(v%3dvs.85)

>You should prepare your applications for this call to fail so that they can either continue without sound or prompt the user to close the application that is already using the sound device.

I am reproducing it by disabling all my sound devices in Device Manager. What I discovered that GM8.1 does is basically retries to create the device, and it will play the game if I enable it in Device Manager. I am going to bring this looping behavior over to ENIGMA since we already have "Retry" and "ignore" on the debug dialog.

**Comment by RobertBColton** [comment]
Hrm, retrying doesn't seem to work with OpenAL because of `OpenAL32.dll` or something.
http://openal.org/pipermail/openal/2016-September/000550.html

**Comment by RobertBColton** [comment]
Alright, finally got the chance to test GMSv1.4 and here's what happens, it's an almost silent fail.
```
No audio output devices found.  Will fallback to use dummy output.
Could not initialize audio output device.  Falling back to use dummy output.
```
